sfxclient/httpsink.go

```diff
@@ -354,6 +372,27 @@ func mapToProperties(properties map[string]interface{}) []*com_signalfx_metrics_
 	return response
 }
 
+type spanResponse struct {
+	Valid   uint64              `json:"valid"`
+	Invalid map[string][]string `json:"invalid"`
+}
+
+func spanResponseValidator(respBody []byte) error {
+	var respObj spanResponse
+
+	if err := json.Unmarshal(respBody, &respObj); err != nil {
+		return errors.Annotatef(err, "cannot unmarshal response body %s", respBody)
+	}
+
+	for _, v := range respObj.Invalid {
+		if len(v) > 0 {
+			return errors.Errorf("some spans were invalid: %v", respObj.Invalid)
```

Comment:
Hrm, this will only return the first kind. Why not implement the error interface that just remarshals map[string][]string ?

Reply:
The error message contains the string representation of the entire `invalid` object from json so it will show everything returned if any one of the invalid fields were present and non-empty.